core/src/main/scala/kafka/server/KafkaRequestHandler.scala

```diff
@@ -21,26 +21,30 @@ import kafka.network._
 import kafka.utils._
 import kafka.metrics.KafkaMetricsGroup
 import java.util.concurrent.{CountDownLatch, TimeUnit}
+import java.util.concurrent.atomic.AtomicInteger
 
 import com.yammer.metrics.core.Meter
 import org.apache.kafka.common.internals.FatalExitError
 import org.apache.kafka.common.utils.{KafkaThread, Time}
 
+import scala.collection.mutable
+
 /**
  * A thread that answers kafka requests.
  */
 class KafkaRequestHandler(id: Int,
                           brokerId: Int,
                           val aggregateIdleMeter: Meter,
-                          val totalHandlerThreads: Int,
+                          val totalHandlerThreads: AtomicInteger,
                           val requestChannel: RequestChannel,
                           apis: KafkaApis,
                           time: Time) extends Runnable with Logging {
```

Comment:
I wonder why we don't use `ShutdownableThread`. Maybe this code predates it.

Reply:
Don't know the answer, didn't change.